File: Assets/Bundles/Client/Resources/TS/Controllers/Global/BlockInteractions/Signal/BeforeBlockHitSignal.lua
-- Compiled with unity-ts v2.1.0-75
local BeforeBlockHitSignal
do
	BeforeBlockHitSignal = setmetatable({}, {
		__tostring = function()
			return "BeforeBlockHitSignal"
		end,
	})
	BeforeBlockHitSignal.__index = BeforeBlockHitSignal
	function BeforeBlockHitSignal.new(...)
		local self = setmetatable({}, BeforeBlockHitSignal)
		return self:constructor(...) or self
	end
	function BeforeBlockHitSignal:constructor(blockPos, block)
		self.blockPos = blockPos
		self.block = block
	end
end
return {
	BeforeBlockHitSignal = BeforeBlockHitSignal,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Bundles/Imports/Core/Core~/src/Server/Services/Block/Signal/BeforeBlockHitSignal.ts
import { Player } from "Shared/Player/Player";
import { Block } from "Shared/VoxelWorld/Block";

export class BeforeBlockHitSignal {
	constructor(
		public readonly block: Block,
		public readonly blockPos: Vector3,
		public readonly player: Player,
		/**
		 * To modify the damage you must edit BlockHitDamageCalc()
		 */
		public readonly damage: number,
		public readonly isGroupEvent: boolean,
	) {}
}

export class BeforeBlockGroupHitSignal {
	constructor(
		public readonly blocks: Block[],
		public readonly blockPositions: Vector3[],
		public readonly player: Player,
		/**
		 * To modify the damage you must edit BlockHitDamageCalc()
		 */
		public readonly damage: number,
	) {}
}
